WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationBaseFormFragment.java

```diff
@@ -48,39 +50,49 @@ public void onCreate(Bundle savedInstanceState) {
 
     protected ViewGroup createMainView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.site_creation_form_screen, container, false);
-        ViewStub formContainer = ((ViewStub) rootView.findViewById(R.id.site_creation_form_content_stub));
+        ViewStub formContainer = (rootView.findViewById(R.id.site_creation_form_content_stub));
         formContainer.setLayoutResource(getContentLayout());
         formContainer.inflate();
         return rootView;
     }
 
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         ViewGroup rootView = createMainView(inflater, container, savedInstanceState);
 
         setupContent(rootView);
 
-        mPrimaryButton = (Button) rootView.findViewById(R.id.primary_button);
-        mSecondaryButton = (Button) rootView.findViewById(R.id.secondary_button);
+        mPrimaryButton = rootView.findViewById(R.id.primary_button);
+        mSecondaryButton = rootView.findViewById(R.id.secondary_button);
         setupBottomButtons(mSecondaryButton, mPrimaryButton);
 
         return rootView;
     }
 
     @Override
-    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
-        Toolbar toolbar = (Toolbar) view.findViewById(R.id.toolbar);
+        Toolbar toolbar = view.findViewById(R.id.toolbar);
         ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
 
         ActionBar actionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
         if (actionBar != null) {
             actionBar.setDisplayHomeAsUpEnabled(true);
-            actionBar.setTitle(R.string.site_creation_title);
+            actionBar.setTitle(getScreenTitle());
```

Comment:
I am really not a big fan of doing this in the base fragment. If the instance creation went through the base fragment, I'd be fine with it, but since we have the `newInstance` in each fragment and the titles are set there, I think I'd have preferred to have it in each fragment even though it's a bit of duplication.

If you feel strongly about keeping it as is, I don't mind it **that much**, so feel free to ignore this comment.

Reply:
Since I want to force the children to set both actionbar title and activity title I made the getScreenTitle method abstract. I hope it makes it cleaner. Let me know if you want me to modify something else.